gtars-igd/src/igdbloom.rs

```diff
@@ -0,0 +1,319 @@
+) -> Result<usize, Box<dyn std::error::Error>> {
+    // Create output directory if it doesn't exist
+    make_parent_directory(output_directory)?;
+
+    let input_path = Path::new(input_directory);
+
+    if !input_path.exists() || !input_path.is_dir() {
+        return Err(format!(
+            "Input directory does not exist or is not a directory: {}",
+            input_directory
+        )
+        .into());
+    }
+
+    let mut processed_count = 0;
+
+    for entry in fs::read_dir(input_path)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        if path.is_file() {
+            if let Some(extension) = path.extension() {
+                if extension == "bed" {
+                    let bed_file = path.to_str().ok_or("Invalid path")?;
+                    println!("Processing BED file: {}", bed_file);
+
+                    tokenize_then_create_bloom_for_each_file(
+                        universe_tokenizer,
+                        bed_file,
+                        output_directory,
+                        num_of_items,
+                        false_positive_rate,
+                    );
+
+                    processed_count += 1;
+                }
+            }
+        }
+    }
+
+    println!(
+        "Processed {} BED files from directory: {}",
+        processed_count, input_directory
+    );
+    Ok(processed_count)
+}
+
+#[cfg(feature = "bloom")]
+pub fn load_bloom_directory(
+    bloom_directory: &str,
+) -> Result<HashMap<String, Bloom<String>>, Box<dyn std::error::Error>> {
+    let bloom_path = Path::new(bloom_directory);
+
+    if !bloom_path.exists() || !bloom_path.is_dir() {
+        return Err(format!(
+            "Bloom directory does not exist or is not a directory: {}",
+            bloom_directory
+        )
+        .into());
+    }
+
+    let mut bloom_filters: HashMap<String, Bloom<String>> = HashMap::new();
+
+    for entry in fs::read_dir(bloom_path)? {
+        let entry = entry?;
+        let path = entry.path();
+
+        if path.is_file() {
+            if let Some(extension) = path.extension() {
+                if extension == "bloom" {
+                    let bloom_file = path.to_str().ok_or("Invalid path")?;
+
+                    let filename = path
+                        .file_stem()
+                        .and_then(|os_str| os_str.to_str())
+                        .ok_or("Invalid filename")?
+                        .to_string();
+
+                    println!("Loading bloom filter: {}", bloom_file);
+
+                    match load_bloom_filter_from_disk(bloom_file) {
+                        Ok(filter) => {
+                            bloom_filters.insert(filename, filter);
+                        }
+                        Err(e) => {
+                            eprintln!("Failed to load bloom filter from {}: {}", bloom_file, e);
+                            // Continue loading other files even if one fails
+                        }
```

Comment:
Do we want to collect this information somewhere, and create log file out of it? Just thoughts